journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournalWriter.java

```diff
@@ -80,34 +84,31 @@ public void append(final JournalRecord record) {
     }
   }
 
-  public void reset(final long index) {
-    currentSegment = journal.resetSegments(index);
+  void reset(final long index) {
+    currentSegment = segments.resetSegments(index);
     currentWriter = currentSegment.writer();
+    flusher.resetLastFlushedIndex(index - 1);
   }
 
-  public void deleteAfter(final long index) {
+  void deleteAfter(final long index) {
     // Delete all segments with first indexes greater than the given index.
-    while (index < currentSegment.index() && currentSegment != journal.getFirstSegment()) {
-      journal.removeSegment(currentSegment);
-      currentSegment = journal.getLastSegment();
+    while (index < currentSegment.index() && currentSegment != segments.getFirstSegment()) {
+      segments.removeSegment(currentSegment);
+      currentSegment = segments.getLastSegment();
       currentWriter = currentSegment.writer();
     }
 
     // Truncate the current index.
     currentWriter.truncate(index);
+    flusher.resetLastFlushedIndex(index - 1);
```

Comment:
:wrench: I think it is better to update it before deleting entries. We didn't do it before also. But what if node crashes after deleting but before updating metastore? There is a small chance the after restart it incorrectly detects journal corruption.

Reply:
If we do it before, then crash before truncating, we would treat any corruption as a partial write. Which is probably fine as we wanted to truncate anyway, I guess :smile: